### PR TITLE
Segment and Cage Card related data migration

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Data transformation and migration of Segment related tables into Protocol related tables. Cage card data migration.

#### Changes
* Cage Card and Cage Card History tables and ETL
* ProtocolCounts, ProtocolProcedures, and ProtocolStress migration from Segment related tables (SEGMENT, SEGMENT_STRESS, SEGMENT_PROCEDURES, SEGMENT_EUTHANASIA)
* ProtocolSupplement multi-step ETL combine some of these protocol ETLs
* Stress table and ETL (STRESS_NAME)
* card_format, census_activity_status, euthanasia, and regulatory_stress_levels TSVs
* Remove Segment references from any migrations and replace with protocol and species
* Cleanup metadata and push location based metadata to NIRC_EHRCustomizer

### CODE REVIEW

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+ALTER TABLE nirc_ehr.AnimalReqOrder DROP COLUMN Segment;
+ALTER TABLE nirc_ehr.AnimalReqOrder ADD COLUMN Protocol integer;
+ALTER TABLE nirc_ehr.AnimalReqOrder ADD COLUMN Species integer;
+
+ALTER TABLE nirc_ehr.Account DROP COLUMN cage;
+ALTER TABLE nirc_ehr.Account DROP COLUMN room;
+ALTER TABLE nirc_ehr.Account DROP COLUMN building;
+
+CREATE TABLE nirc_ehr.ProtocolStress
```
Comment (labkey-jeckels) [inline nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql:12]: We generally have trouble with tables that don't have a PK. I suggest adding a RowId if there's not a natural PK in the data itself.

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    Created                 TIMESTAMP,
+    CreatedBy               USERID,
+    Modified                TIMESTAMP,
+    ModifiedBy              USERID,
+    CONSTRAINT PK_CAGECARD PRIMARY KEY (CageCardId),
+    CONSTRAINT FK_CAGECARD_Container FOREIGN KEY (Container) REFERENCES core.Containers (EntityId)
+);
+CREATE INDEX IX_Nirc_Ehr_Cage_Card_Container ON nirc_ehr.CageCard (Container);
+
+CREATE TABLE nirc_ehr.CageCardHistory
```
Comment (labkey-jeckels) [inline nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql:94]: Same here - can we add a PK?

**Review by labkey-jeckels — APPROVED** [review]

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql (hunk 1 above)
Comment (labkey-martyp) [reply nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql:12]: Ok added RowId

File: nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql (hunk 2 above)
Comment (labkey-martyp) [reply nirc_ehr/resources/schemas/dbscripts/postgresql/nirc_ehr-22.007-22.008.sql:94]: added

**Pull request merged**